src/Carousel.tsx

```diff
@@ -0,0 +1,80 @@
+import * as React from 'react';
+import * as ReactDOM from 'react-dom';
+import { HistoryActions } from './App.tsx';
+import { IAttachment, Attachment } from './Attachment.tsx';
+
+export interface Props {
+    attachments: any[];
+    actions: HistoryActions;
+}
+
+export interface State {
+    scrollIndex: number;
+    itemWidth: number;
+    ul?: HTMLUListElement
+    scrollDiv?: HTMLDivElement;
+}
+
+export class Carousel extends React.Component<Props, State> {
+
+    constructor(props: Props) {
+        super();
+
+        this.state = {
+            scrollIndex: 0,
+            itemWidth: null
+        };
+
+    }
+
+    private componentDidMount() {
+        var li = this.state.ul.firstChild as HTMLLIElement;
+        this.state.itemWidth = li.offsetWidth;
+    }
+
+    private scrollBy(increment: number) {
```

Comment:
Generally I'm trying to leverage the native scrolling and not reinvent it. I'm using the scrollLeft property which is driven by resize. This is still being worked out.